src/api/client.js

```diff
@@ -0,0 +1,104 @@
+// import axios and create pre-configured version of axios
+import axios from 'axios'
+import { mutate } from 'swr';
+
+const client = axios.create({
+  baseURL: 'http://localhost:3000'
+})
+
+export default client
+
+/**
+ * @typedef PayloadWithId
+ * @type {object}
+ * @property {string} id - The ID of the payload
+ */
+
+/**
+ * Creates an entity associated with the current output
+ * @param {string} endpoint the endpoint that handles the entity
+ * @param {object} payload the payload for the object
+ * @returns {Array} A list containing the response and error respectivly
+ */
+export async function createEntity(endpoint, payload) {
```

Comment:
For the parameters of the createEntity function, besides endpoint and payload, should we also allow a third parameter like config in case the client code requires additional config of the axios request? For example, specifying content-type header, etc.